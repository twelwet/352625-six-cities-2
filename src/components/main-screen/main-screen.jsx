import React from "react";
import Cities from "../cities/cities.jsx";
import Offers from "../offers/offers.jsx";
import MapComponent from "../map/map.jsx";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer.js";

const MainScreen = (props) => {
  return <main className="page__main page__main--index">
    <h1 className="visually-hidden">Cities</h1>
    <div className="tabs">
      <Cities {...props}/>
    </div>
    <div className="cities">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">312 places to stay in Amsterdam</b>
          <form className="places__sorting" action="#" method="get">
            <span className="places__sorting-caption">Sort by</span>
            <span className="places__sorting-type" tabIndex="0">
              Popular
              <svg className="places__sorting-arrow" width="7" height="4">
                <use xlinkHref="#icon-arrow-select"/>
              </svg>
            </span>
            <ul className="places__options places__options--custom places__options--opened">
              <li className="places__option places__option--active" tabIndex="0">Popular</li>
              <li className="places__option" tabIndex="0">Price: low to high</li>
              <li className="places__option" tabIndex="0">Price: high to low</li>
              <li className="places__option" tabIndex="0">Top rated first</li>
            </ul>
          </form>
          <div className="cities__places-list places__list tabs__content">
            <Offers {...props}/>
          </div>
        </section>
        <div className="cities__right-section">
          <section className="cities__map map">
            <MapComponent {...props}/>
          </section>
        </div>
      </div>
    </div>
  </main>;
};

MainScreen.propTypes = {
  citiesList: PropTypes.arrayOf(PropTypes.string),
  city: PropTypes.string,
  cityOffers: PropTypes
    .arrayOf(PropTypes
      .exact({
        id: PropTypes.number.isRequired,
        city: PropTypes.exact({
          location: PropTypes.exact({
            name: PropTypes.string.isRequired,
            latitude: PropTypes.number.isRequired,
            longitude: PropTypes.number.isRequired,
            zoom: PropTypes.number.isRequired
          })
        }),
        location: PropTypes.exact({
          latitude: PropTypes.number.isRequired,
          longitude: PropTypes.number.isRequired,
          zoom: PropTypes.number.isRequired
        }),
        description: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        price: PropTypes.number.isRequired,
        image: PropTypes.string.isRequired,
        rating: PropTypes.number.isRequired,
        isPremium: PropTypes.bool.isRequired,
        isBookmark: PropTypes.bool.isRequired,
      })
    ),
  onCityClick: PropTypes.func
};

const mapStateToProps = (state) => ({
  citiesList: state.citiesList,
  city: state.city,
  cityOffers: state.cityOffers,
  offers: state.offers
});

const mapDispatchToProps = (dispatch) => ({
  onCityClick: (city) => {
    dispatch(ActionCreator.changeCity(city));
    dispatch(ActionCreator.getOffers(city));
  }
});

export {MainScreen};

export default connect(mapStateToProps, mapDispatchToProps)(MainScreen);

