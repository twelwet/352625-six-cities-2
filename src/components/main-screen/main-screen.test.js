import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16/build";
import toJson from "enzyme-to-json";
import {MainScreen} from "./main-screen.jsx";

const mockOffers = [
  {
    [`id`]: 111,
    [`city`]: {
      [`name`]: `City Name`,
      [`location`]: {
        [`latitude`]: 52.38333,
        [`longitude`]: 4.9,
        [`zoom`]: 12
      }
    },
    [`location`]: {
      [`latitude`]: 52.3909553943508,
      [`longitude`]: 4.85309666406198,
      [`zoom`]: 12
    },
    [`title`]: `Some text`,
    [`type`]: `Apartment`,
    [`price`]: 120,
    [`preview_image`]: `img/img-1.jpg`,
    [`rating`]: 100,
    [`is_premium`]: true,
    [`is_favorite`]: false
  },
  {
    [`id`]: 222,
    [`city`]: {
      [`name`]: `City Name 2`,
      [`location`]: {
        [`latitude`]: 52.38333,
        [`longitude`]: 4.9,
        [`zoom`]: 12
      }
    },
    [`location`]: {
      [`latitude`]: 52.3909553943508,
      [`longitude`]: 4.85309666406198,
      [`zoom`]: 12
    },
    [`title`]: `Some text`,
    [`type`]: `House`,
    [`price`]: 120,
    [`preview_image`]: `img/img-2.jpg`,
    [`rating`]: 100,
    [`is_premium`]: false,
    [`is_favorite`]: true
  }
];

Enzyme.configure({adapter: new Adapter()});

it(`MainScreen correctly renders after relaunch`, () => {
  const onCityClick = jest.fn();
  const onBookmarkClick = jest.fn();

  const tree = shallow(<MainScreen
    citiesList={[`Amsterdam`, `Berlin`, `Helsinki`]}
    city={`Amsterdam`}
    cityOffers={mockOffers}
    onCityClick={onCityClick}
    onBookmarkClick={onBookmarkClick}
    wrapperClass={`wrapper-class`}
    offersClass={`offers-class`}
    offerArticleClass={`offer-article`}
    offerDivClass={`offer-div`}
  />);

  expect(toJson(tree)).toMatchSnapshot();
});
