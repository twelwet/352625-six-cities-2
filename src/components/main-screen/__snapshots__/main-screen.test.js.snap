// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen correctly renders after relaunch 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <WithActiveElement
      citiesList={
        Array [
          "Amsterdam",
          "Berlin",
          "Helsinki",
        ]
      }
      city="Amsterdam"
      cityOffers={
        Array [
          Object {
            "city": Object {
              "location": Object {
                "latitude": 52.38333,
                "longitude": 4.9,
                "zoom": 12,
              },
              "name": "City Name",
            },
            "id": 111,
            "is_favorite": false,
            "is_premium": true,
            "location": Object {
              "latitude": 52.3909553943508,
              "longitude": 4.85309666406198,
              "zoom": 12,
            },
            "preview_image": "img/img-1.jpg",
            "price": 120,
            "rating": 100,
            "title": "Some text",
            "type": "Apartment",
          },
          Object {
            "city": Object {
              "location": Object {
                "latitude": 52.38333,
                "longitude": 4.9,
                "zoom": 12,
              },
              "name": "City Name 2",
            },
            "id": 222,
            "is_favorite": true,
            "is_premium": false,
            "location": Object {
              "latitude": 52.3909553943508,
              "longitude": 4.85309666406198,
              "zoom": 12,
            },
            "preview_image": "img/img-2.jpg",
            "price": 120,
            "rating": 100,
            "title": "Some text",
            "type": "House",
          },
        ]
      }
      offerArticleClass="offer-article"
      offerDivClass="offer-div"
      offersClass="offers-class"
      onBookmarkClick={[MockFunction]}
      onCityClick={[MockFunction]}
      wrapperClass="wrapper-class"
    />
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          2
           places to stay in 
          Amsterdam
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            className="places__sorting-type"
            tabIndex="0"
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom places__options--opened"
          >
            <li
              className="places__option places__option--active"
              tabIndex="0"
            >
              Popular
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Price: low to high
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Price: high to low
            </li>
            <li
              className="places__option"
              tabIndex="0"
            >
              Top rated first
            </li>
          </ul>
        </form>
        <WithActiveElement
          citiesList={
            Array [
              "Amsterdam",
              "Berlin",
              "Helsinki",
            ]
          }
          city="Amsterdam"
          cityOffers={
            Array [
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.38333,
                    "longitude": 4.9,
                    "zoom": 12,
                  },
                  "name": "City Name",
                },
                "id": 111,
                "is_favorite": false,
                "is_premium": true,
                "location": Object {
                  "latitude": 52.3909553943508,
                  "longitude": 4.85309666406198,
                  "zoom": 12,
                },
                "preview_image": "img/img-1.jpg",
                "price": 120,
                "rating": 100,
                "title": "Some text",
                "type": "Apartment",
              },
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.38333,
                    "longitude": 4.9,
                    "zoom": 12,
                  },
                  "name": "City Name 2",
                },
                "id": 222,
                "is_favorite": true,
                "is_premium": false,
                "location": Object {
                  "latitude": 52.3909553943508,
                  "longitude": 4.85309666406198,
                  "zoom": 12,
                },
                "preview_image": "img/img-2.jpg",
                "price": 120,
                "rating": 100,
                "title": "Some text",
                "type": "House",
              },
            ]
          }
          offerArticleClass="offer-article"
          offerDivClass="offer-div"
          offersClass="offers-class"
          offersList={
            Array [
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.38333,
                    "longitude": 4.9,
                    "zoom": 12,
                  },
                  "name": "City Name",
                },
                "id": 111,
                "is_favorite": false,
                "is_premium": true,
                "location": Object {
                  "latitude": 52.3909553943508,
                  "longitude": 4.85309666406198,
                  "zoom": 12,
                },
                "preview_image": "img/img-1.jpg",
                "price": 120,
                "rating": 100,
                "title": "Some text",
                "type": "Apartment",
              },
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.38333,
                    "longitude": 4.9,
                    "zoom": 12,
                  },
                  "name": "City Name 2",
                },
                "id": 222,
                "is_favorite": true,
                "is_premium": false,
                "location": Object {
                  "latitude": 52.3909553943508,
                  "longitude": 4.85309666406198,
                  "zoom": 12,
                },
                "preview_image": "img/img-2.jpg",
                "price": 120,
                "rating": 100,
                "title": "Some text",
                "type": "House",
              },
            ]
          }
          onBookmarkClick={[MockFunction]}
          onCityClick={[MockFunction]}
          wrapperClass="wrapper-class"
        />
      </section>
      <div
        className="cities__right-section"
      >
        <MapComponent
          citiesList={
            Array [
              "Amsterdam",
              "Berlin",
              "Helsinki",
            ]
          }
          city="Amsterdam"
          cityOffers={
            Array [
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.38333,
                    "longitude": 4.9,
                    "zoom": 12,
                  },
                  "name": "City Name",
                },
                "id": 111,
                "is_favorite": false,
                "is_premium": true,
                "location": Object {
                  "latitude": 52.3909553943508,
                  "longitude": 4.85309666406198,
                  "zoom": 12,
                },
                "preview_image": "img/img-1.jpg",
                "price": 120,
                "rating": 100,
                "title": "Some text",
                "type": "Apartment",
              },
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.38333,
                    "longitude": 4.9,
                    "zoom": 12,
                  },
                  "name": "City Name 2",
                },
                "id": 222,
                "is_favorite": true,
                "is_premium": false,
                "location": Object {
                  "latitude": 52.3909553943508,
                  "longitude": 4.85309666406198,
                  "zoom": 12,
                },
                "preview_image": "img/img-2.jpg",
                "price": 120,
                "rating": 100,
                "title": "Some text",
                "type": "House",
              },
            ]
          }
          mapClass="cities__map map"
          mapHeight="100%"
          offerArticleClass="offer-article"
          offerDivClass="offer-div"
          offersClass="offers-class"
          offersList={
            Array [
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.38333,
                    "longitude": 4.9,
                    "zoom": 12,
                  },
                  "name": "City Name",
                },
                "id": 111,
                "is_favorite": false,
                "is_premium": true,
                "location": Object {
                  "latitude": 52.3909553943508,
                  "longitude": 4.85309666406198,
                  "zoom": 12,
                },
                "preview_image": "img/img-1.jpg",
                "price": 120,
                "rating": 100,
                "title": "Some text",
                "type": "Apartment",
              },
              Object {
                "city": Object {
                  "location": Object {
                    "latitude": 52.38333,
                    "longitude": 4.9,
                    "zoom": 12,
                  },
                  "name": "City Name 2",
                },
                "id": 222,
                "is_favorite": true,
                "is_premium": false,
                "location": Object {
                  "latitude": 52.3909553943508,
                  "longitude": 4.85309666406198,
                  "zoom": 12,
                },
                "preview_image": "img/img-2.jpg",
                "price": 120,
                "rating": 100,
                "title": "Some text",
                "type": "House",
              },
            ]
          }
          onBookmarkClick={[MockFunction]}
          onCityClick={[MockFunction]}
          wrapperClass="wrapper-class"
        />
      </div>
    </div>
  </div>
</main>
`;
